### PR TITLE
enh: throw dust error for known notion transient errors

### PR DESCRIPTION
**Pull request by fontanierh**

### CODE REVIEW

File: connectors/src/connectors/notion/temporal/cast_known_errors.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    } catch (err: unknown) {
+      if (APIResponseError.isAPIResponseError(err)) {
+        if (err.code === APIErrorCode.RateLimited) {
+          throw {
+            __is_dust_error: true,
+            message: err.message,
+            type: "notion_rate_limited",
+          };
+        }
+        if (err.code === APIErrorCode.InternalServerError) {
```
Comment (spolu) [inline connectors/src/connectors/notion/temporal/cast_known_errors.ts:26]: What are these errors?

**Review by spolu — APPROVED** [review]
Left a small question but otherwise LGTM!

File: connectors/src/connectors/notion/temporal/cast_known_errors.ts (hunk 1 above)
Comment (spolu) [reply connectors/src/connectors/notion/temporal/cast_known_errors.ts:26]: The real meta question, is should we page ourselves if they happen?

Comment (fontanierh) [reply connectors/src/connectors/notion/temporal/cast_known_errors.ts:26]: The ones we regularly have are transient internal server errors that on the search endpoint for big workspaces. Technically, they could be anything

Comment (spolu) [reply connectors/src/connectors/notion/temporal/cast_known_errors.ts:26]: OK if regular, we should not page ourselves and therefore this is good to go :+1:

**Pull request merged**